polar-core/src/warning.rs

```diff
@@ -0,0 +1,149 @@
+use std::fmt;
+
+use indoc::indoc;
+
+use super::diagnostic::{Context, Range};
+use super::kb::KnowledgeBase;
+use super::terms::{InstanceLiteral, Pattern, Symbol, Term, Value};
+
+#[derive(Debug)]
+pub struct PolarWarning {
```

Comment:
If this type has the same structure as `Error` why is it a different type? Is this the intended final state, or just a stepping stone?

Reply:
Stepping stone. All we need for the current LSP work is enough structure on warnings to translate them into VSCode diagnostics. After #1368, we have enough structure on errors to translate them into VSCode diagnostics, so this PR just aligns warnings with errors. I think we can deal with the duplication (via a trait or something) in some future work on diagnostics that isn't specific to the LSP project. Sound good?